runtime/metrics/src/lib.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Runtime metric interface similar to native Prometheus metrics.
+//!
+//! This is intended to be used only for testing and debugging and **must never
+//! be used in production**. It requires the Substrate wasm tracing support
+//! and command line configuration: `--tracing-targets wasm_tracing=trace`.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(feature = "std")]
+mod std;
+#[cfg(feature = "std")]
+pub use crate::std::*;
+
+#[cfg(not(feature = "std"))]
+mod nostd;
+#[cfg(not(feature = "std"))]
+pub use crate::nostd::*;
```

Comment:
This should be changed.

This code currently means, even if we don't have `runtime-metrics` enabled, we will use the `nostd` version that is not a `no-op` when `runtime-metrics` are disabled.

The files should be renamed to `with_runtime_metrics` and `without_runtime_metrics` to reflect this better and then the file should be chose based on the `runtime-metrics` feature.

The code should be a no-op for production.

Reply:
fixed